eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/ChannelTlv.scala

```diff
@@ -34,14 +34,25 @@ object ChannelTlv {
     val isEmpty: Boolean = script.isEmpty
   }
 
+  val upfrontShutdownScriptCodec: Codec[UpfrontShutdownScript] = variableSizeBytesLong(varintoverflow, bytes).as[UpfrontShutdownScript]
+
+  /** A channel type is a set of even feature bits that represent persistent features which affect channel operations. */
+  case class ChannelType(features: Features) extends OpenChannelTlv with AcceptChannelTlv
+
+  val channelTypeCodec: Codec[ChannelType] = variableSizeBytesLong(varintoverflow, bytes).xmap(
+    b => ChannelType(Features(b)),
+    ct => ct.features.toByteVector
+  )
```

Comment:
I think we should:
- suffix tlvs by `Tlv` as a convention, this is consistent with the naming of traits, and removes conflicts between `ChannelTlv.ChannelType`/`channel.ChannelType`
- not expose the underlying `features`, which imo are a serialization detail
- make `ChannelTypes.fromFeatures` return a `Try` (this would make `Attempt.fromOption` move to `Attempt.fromTry`) (*)

```suggestion
  case class ChannelTypeTlv(channelType: ChannelType) extends OpenChannelTlv with AcceptChannelTlv

  val channelTypeCodec: Codec[ChannelTypeTlv] = variableSizeBytesLong(varintoverflow, bytes).exmap(
    b => Attempt.fromOption(ChannelTypes.fromFeatures(Features(b)).map(ChannelTypeTlv), ifNone = Err(s"unknown channel type with features=${Features(b)}")),
    ct => Attempt.successful(ct.channelType.features.toByteVector)
  )
```

(*) on 2nd thought: an unknown channel type will trigger a deserialization error and a connection close. The spec requires a *channel close* only:
> The receiving node MUST fail the channel if:
> - It supports `channel_type`, `channel_type` was set, and the `type` is not suitable.

(by the way there is an ambiguity between *fail* and *reject* the channel in the spec)

I think a cleaner approach would be to define an `UnsupportedChannelType` along `Standard`/`StaticRemoteKey`/`AnchorOutputs`. There could still be deserialization errors if, for example, odd bits are used, but in case it's just an unknown channel type, we could handle it properly. Just ignoring unknown channel types doesn't work because then we can't discriminate between "unknown channel type" and "no channel type".

Reply:
> I think a cleaner approach would be to define an UnsupportedChannelType along Standard/StaticRemoteKey/AnchorOutputs

Good idea, I'll try that.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1867/commits/03bf4c363efa65184c39a2af079fb8976f389b77